impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js

```diff
@@ -174,6 +176,17 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
               this.popup.set('title',this.title);
             },
 
+            setupResponsiveness: function () {
+              //add "responsive" class to the dialog, along with any additional responsiveClasses defined
+              domClass.add(this.popup.domNode, "responsive");
+              if (this.responsiveClasses) {
+                domClass.add(this.popup.domNode, this.responsiveClasses);
+              }
+
+              //move the titlebar to the dialog content to take advantage of the responsive CSS changes we've made in PAZ
+              construct.place(query('.dijitDialogTitleBar',this.popup.domNode)[0], query('.dijitDialogPaneContent',this.popup.domNode)[0], "first");
```

Comment:
Since the buttons also have to be siblings of the content and not a child thereof, and the buttons are not created until `show()`, and `show()` has a `shown` flag existing and being used, I have decided to `_setupResonsiveness` there, using the existing flag/conditional